src/Idris/Desugar.idr

```diff
@@ -694,14 +694,14 @@ mutual
            pure $ MkImpTy fc nameFC n !(bindTypeNames fc (usingImpl syn)
                                                ps !(desugar AnyExpr ps ty))
 
+  -- Attempt to get function name from function pattern. For example, given the
+  -- pattern 'f x y', getClauseFn would return 'f'.
   getClauseFn : RawImp -> Core Name
   getClauseFn (IVar _ n) = pure n
   getClauseFn (IApp _ f _) = getClauseFn f
   getClauseFn (IAutoApp _ f _) = getClauseFn f
   getClauseFn (INamedApp _ f _ _) = getClauseFn f
-  getClauseFn tm = throw $ case tm of
-    Implicit fc _ => GenericMsg fc "Invalid name for a declaration"
-    _ => InternalError (show tm ++ " is not a function application")
+  getClauseFn tm = throw $ GenericMsg (getFC tm) "Leftmost term in pattern must be a function name"
```

Comment:
It's not the leftmost one but the head one. Patterns on the LHS can you use infix
names, e.g. `(x :: xs) !! Z = x`

Reply:
Ah right, I'll change that now

Reply:
Ok I've fixed the wording. I also added a comment above the `getClauseFn` to try and help other new contributors, however it's probably worth verifying the accuracy of what I've written.